src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -306,44 +327,68 @@ private int ParseMultiSpanHeader(TRequestHandler handler, ref SequenceReader<byt
 
         SequencePosition lineEnd;
         scoped ReadOnlySpan<byte> headerSpan;
-        if (currentSlice.Slice(reader.Position, lineEndPosition.Value).Length == currentSlice.Length - 1)
+        ReadOnlySequence<byte> header;
+        if (_disableHttp1LineFeedTerminators && (currentSlice.Slice(reader.Position, lineEndPosition.Value).Length == currentSlice.Length - 1))
```

Comment:
Even in the case where we allow LF terminators, if a CR arrives first and we don't have the LF, we need to short circuit and return -1 because we cannot parse the entire header. I think the current code risks throwing an `ArgumentOutOfRangeException` from `currentSlice.GetPosition(1, firstLineEndCharPos);`.